pkg/cmd/repo/delete/http.go

```diff
@@ -0,0 +1,38 @@
+package delete
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+func deleteRepo(client *http.Client, repo ghrepo.Interface) error {
+	url := fmt.Sprintf("%srepos/%s",
+		ghinstance.RESTPrefix(repo.RepoHost()),
+		ghrepo.FullName(repo))
+
+	request, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(request)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	err = api.HandleHTTPError(resp)
+
+	if resp.StatusCode == 403 {
+		return fmt.Errorf(`%w
+Try authorizing the "delete_repo" scope with "gh auth refresh -s delete_repo".`, err)
+	} else if resp.StatusCode > 299 {
+		return err
+	}
```

Comment:
Please have the error handling be like this:
```suggestion
	if resp.StatusCode > 299 {
		return api.HandleHTTPError(resp)
	}
```
@vilmibm: We do not have to add special code to warn about missing OAuth scopes anymore https://github.com/cli/cli/pull/4513. HandleHTTPError now does this automatically, informed by the response from the server:
```
gh: Must have admin rights to Repository. (HTTP 403)
gh: This API operation needs the "delete_repo" scope. To request it, run:  gh auth refresh -h github.com -s delete_repo
```

Reply:
I don't think this is working as you expect or I'm doing something wrong because this is my experience: 

<img width="711" alt="Screen Shot 2021-10-15 at 15 25 02" src="https://user-images.githubusercontent.com/98482/137549243-8aa489b0-7c07-46cb-ab3e-ef835af5e11e.png">

Reply:
ah, my apologies. this PR hasn't been merged with trunk recently; I see the scope refresh error now